src/modules/keepalive/keepalive_mod.c

```diff
@@ -235,24 +244,61 @@ static int ki_del_destination(sip_msg_t *msg, str *uri, str *owner)
 
 /*
  * Function callback executer per module param "destination".
- * Is just a wrapper to ka_add_dest() api function
+ * It just adds destinations to an initial list to be added later in mod_init
+ * This is required because of initialization requirements.
  */
-static int ka_mod_add_destination(modparam_t type, void *val)
-{
-	if(ka_alloc_destinations_list() < 0)
-		return -1;
+static int ka_mod_add_destination(modparam_t type, void *val) {
+	LM_DBG("adding destination to initial list %s\n", (char *)val);
+
+	char *owner = "_params";
+	char *uri = (char *)val;
+
+	ka_initial_dest_t * new_destination = (ka_initial_dest_t *) shm_malloc(sizeof(ka_initial_dest_t));
+	new_destination->uri.s = shm_malloc(sizeof(char) * strlen(uri));
+	new_destination->owner.s = shm_malloc(sizeof(char) * strlen(owner));
+
+	memcpy(new_destination->uri.s, uri, strlen(uri));
+	new_destination->uri.len = strlen(uri);
+
+	memcpy(new_destination->owner.s, owner, strlen(owner));
+	new_destination->owner.len = strlen(owner);
+	
+	new_destination->next = NULL;
+
+	if (ka_initial_destinations_list == NULL) {
+		ka_initial_destinations_list = new_destination;
+	} else {
+		ka_initial_dest_t *current_position = ka_initial_destinations_list;
```

Comment:
current_position has to be defined at the beginning of the function

Reply:
Is there any programming style recommendation? I'm sorry to disagree here but some code checkers may complain about that saying "The scope of the variable 'x' can be reduced"

Reply:
Just an example: https://github.com/kamailio/kamailio/commit/daa86b204d3030e2abb2f2459aa348998b982ae2

Reply:
I still see more cons than pros, but OK. Updating this.